src/Daemon/BaseDaemon.cpp

```diff
@@ -837,7 +837,12 @@ void BaseDaemon::initialize(Application & self)
         }
     }
 
-    /// This must be done before any usage of DateLUT. In particular, before any logging.
+    /// These two parameters must be processed before any usage of DateLUT. In particular, before any logging.
+    if (config().has("prefer_system_tzdata"))
```

Comment:
Before this PR zone_info_source_factory is set regardless if BaseDaemon::initialize was called or not.

So the concern is 
1. do we have application (clickhouse-local ?) that bypasses BaseDaemon::initialize ?
2. if yes, is it a problem ?

Reply:
Looks like ch-local does not respect this parameter indeed.
Idk if this is a major problem -- we already have config parameters that are not respected by ch-local. E.g. take a look at the `timezone` parameter that is processed below

Reply:
I never really used CH-local, but for me it also looks like it doesn't respect logging settings that are specified in `config.xml`

Reply:
ch-local definitely operates with dates in general.
How it would work if DateLUT::setTZDataSource is not called?

Reply:
Actually, this is an issue, you're right.
It will use system tzdata then.

Reply:
Fixed now